homeassistant/components/systemmonitor/sensor.py

```diff
@@ -592,172 +545,35 @@ def __init__(
             name="System Monitor",
         )
 
-    @property
-    def native_value(self) -> str | datetime | None:
-        """Return the state of the device."""
-        return self.data.state
-
-    @property
-    def available(self) -> bool:
-        """Return True if entity is available."""
-        return self.data.last_exception is None
-
-    @property
-    def data(self) -> SensorData:
-        """Return registry entry for the data."""
-        return self._sensor_registry[(self.entity_description.key, self._argument)]
-
     async def async_added_to_hass(self) -> None:
-        """When entity is added to hass."""
+        """When entity is added."""
         await super().async_added_to_hass()
-        self.async_on_remove(
-            async_dispatcher_connect(
-                self.hass, SIGNAL_SYSTEMMONITOR_UPDATE, self.async_write_ha_state
-            )
-        )
+        await self.coordinator.async_request_refresh()
```

Comment:
I'm not sure we want to request a refresh here. It would mean that there will be at least a second refresh after the cooldown time (10 seconds) if more than one entity shares the same coordinator.

Probably refresh the coordinators before creating the entities and then pass each entity its coordinator.